include/yaclib/coroutine/await.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <yaclib/async/future.hpp>
+#include <yaclib/util/detail/atomic_counter.hpp>
+#include <yaclib/util/type_traits.hpp>
+
+#include <tuple>
+
+#if defined(__clang__)
+
+#include <experimental/coroutine>
+namespace yaclib_std {
+using std::experimental::coroutine_handle;
+}  // namespace yaclib_std
+
+#else
+
+#include <coroutine>
+
+namespace yaclib_std {
+using std::coroutine_handle;
+}  // namespace yaclib_std
+
+#endif
+
+namespace yaclib {
+
+class FutureAwaiter final {
+ public:
+  FutureAwaiter() = delete;
+  FutureAwaiter(const FutureAwaiter&) = delete;
+  FutureAwaiter(FutureAwaiter&&) = delete;
+  FutureAwaiter& operator=(FutureAwaiter&&) = delete;
+  FutureAwaiter& operator=(const FutureAwaiter&) = delete;
+  ~FutureAwaiter() = default;
+
+  template <typename... V, typename... E>
+  explicit FutureAwaiter(Future<V, E>&... fs) : _future_awaiter_core{1 + sizeof...(fs)} {
```

Comment:
```suggestion
  explicit FutureAwaiter(Future<V, E>&... fs) : _await_core{1 + sizeof...(fs)} {
```
Like `wait_core` in `Wait` function